src/windows/wslaservice/inc/wslaservice.idl

```diff
@@ -116,9 +204,41 @@ struct _WSLA_SESSION_SETTINGS {
 ]
 interface IWSLASession : IUnknown
 {
+    // Image management.
+    HRESULT PullImage([in] LPCWSTR Image, [in, unique] struct WSLA_REGISTRY_AUTHENTICATION_INFORMATION* RegistryInformation, [in, unique] IProgressCallback* ProgressCallback);
+    HRESULT ListImages([out, size_is(, *Count)] struct WSLA_IMAGE_INFORMATION** Images, [out] ULONG* Count);
```

Comment:
Does the callee allocate the image array? How does it get freed?

Reply:
COM handles the array allocation for the caller, but the caller is responsible for freeing it. 

Here's an example on how this works in a similar WSL API: https://github.com/microsoft/WSL/blob/09c17eb6c6f7b678654bccd59ab7ad060dc2192e/src/windows/common/svccomm.cpp#L592C3-L593C133